### PR TITLE
Upgrade pac4j to version 5.1.3

### PR DESCRIPTION
**Pull request by leszekgruchala**

### CODE REVIEW

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 1]
```diff
@@ -72,34 +71,35 @@ object AkkaHttpSecurity {
...
-) {
+class AkkaHttpSecurity(config: Config,
+                       sessionStorage: SessionStorage,
+                       sessionStore: SessionStore,
+                       val sessionCookieName: String = AkkaHttpWebContext.DEFAULT_COOKIE_NAME)(implicit val executionContext: ExecutionContext) {
 
   import AkkaHttpSecurity._
 
-  private[pac4j] val securityLogic: AkkaHttpSecurityLogic =
+  private[pac4j] val securityLogic: SecurityLogic =
```
Comment (ivanopagano) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala:81]: Nice low-hanging-fruit you got here!

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala [hunk 2]
```diff
@@ -129,22 +131,24 @@ class AkkaHttpSecurity(config: Config, sessionStorage: SessionStorage, val sessi
...
         // TODO This is a hack to ensure that any underlying Futures are scheduled (and handled in case of errors) from here
         // TODO Fix this properly
         Future.successful(()).flatMap { _ =>
-          val securityGrantedAccessAdapter = new SecurityGrantedAccessAdapter[Future[RouteResult], AkkaHttpWebContext] {
-            override def adapt(context: AkkaHttpWebContext,
-                               profiles: util.Collection[UserProfile],
-                               @unused parameters: AnyRef*): Future[RouteResult] = {
-              val authenticatedRequest = AuthenticatedRequest(context, profiles.asScala.toList)
+          val securityGrantedAccessAdapter: SecurityGrantedAccessAdapter = new SecurityGrantedAccessAdapter {
+            @unused override def adapt(context: WebContext,
```
Comment (ivanopagano) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala:141]: What's the @unused tag, now?

File: src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala [hunk 3]
```diff
@@ -82,12 +81,15 @@ case class AkkaHttpWebContext(request: HttpRequest,
     changes = changes.copy(cookies = changes.cookies ++ List(httpCookie))
   }
 
-  override lazy val getSessionStore = new AkkaHttpSessionStore()
+  val getSessionStore = sessionStore
```
Comment (ivanopagano) [inline src/main/scala/com/stackstate/pac4j/AkkaHttpWebContext.scala:84]: Is this actually needed if it doesn't override anything now?

File: src/main/scala/com/stackstate/pac4j/AkkaWebContextUtils.scala [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.stackstate.pac4j
+
+import org.pac4j.core.context.WebContext
+
+object AkkaWebContextUtils {
+
+  def withAkkaContext[A](context: WebContext)(f: AkkaHttpWebContext => A): A = {
+    context match {
+      case akka: AkkaHttpWebContext => f(akka)
+      case _ => throw new UnsupportedOperationException("Non Akka WebContext is not supported")
```
Comment (ivanopagano) [inline src/main/scala/com/stackstate/pac4j/AkkaWebContextUtils.scala:10]: Can we rephrase, please? Isn't that sentence confusing in a log mess?
```suggestion
      case _ => throw new UnsupportedOperationException("Only Akka Http WebContext is actually supported")
```

File: src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala [hunk 5]
```diff
@@ -1,56 +1,51 @@
...
-      case _: NoContentAction =>
-        HttpResponse(NoContent)
-      case _ if action.getCode == 500 =>
-        HttpResponse(InternalServerError)
-      case _ =>
-        HttpResponse(StatusCodes.getForKey(action.getCode).getOrElse(custom(action.getCode, "")))
-    }))
+object AkkaHttpActionAdapter extends HttpActionAdapter {
+
+  override def adapt(action: HttpAction, context: WebContext): Future[Complete] = {
```
Comment (ivanopagano) [inline src/main/scala/com/stackstate/pac4j/http/AkkaHttpActionAdapter.scala:16]: Isn't this eventually equivalent to
```scala
override def adapt(action: HttpAction, context: WebContext): Future[Complete] =
  AkkaWebContextUtils.withAkkaContext(context) { akka =>
    Future.successful(Complete(action match {
      case ...
      case ...
  }
```
?

**Review by ivanopagano — COMMENTED** [review]
Started leaving some comment

File: src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala (hunk 2 above)
Comment (leszekgruchala) [reply src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala:141]: To please the compiler

Comment (ivanopagano) [reply src/main/scala/com/stackstate/pac4j/AkkaHttpSecurity.scala:141]: The compiler's our tyrant

**Review by ivanopagano — APPROVED** [review]

**Pull request merged**